bindings/python/pymongoarrow/lib.pyx

```diff
@@ -243,10 +241,13 @@ def process_bson_stream(bson_stream, context, arr_value_builder=None):
                     if value_t == BSON_TYPE_UTF8:
                         bson_str = bson_iter_utf8(&doc_iter, &str_len)
                         builder.append(<bytes>(bson_str)[:str_len])
-                    elif value_t == BSON_TYPE_DECIMAL128:
+                    else:
+                        builder.append_null()
+                elif ftype == t_decimal128:
+                    if value_t == BSON_TYPE_DECIMAL128:
                         bson_iter_decimal128(&doc_iter, &dec128)
-                        bson_decimal128_to_string(&dec128, decimal128_str)
-                        builder.append(<bytes>(decimal128_str))
+                        val = pstruct.pack('=QQ', dec128.low, dec128.high)
```

Comment:
I think this is right on little endian machines where the format should be `('<QQ', dec128.low, dec128.high)` but are you sure this is the correct byte order on big endian? [Based on libbson's bson_decimal128_t struct definition](http://mongoc.org/libbson/current/bson_decimal128_t.html) I would expect the format to be `('>QQ', dec128.high, dec128.low)`

Reply:
Also whatever the outcome here, the Pandas decoding via Decimal128.from_bid expects to receive the little endian format.

Reply:
Updated to correct for system endianness.